### PR TITLE
Uf 5848 update persistence layer

### PR DESCRIPTION
**Pull request by mha109**

### CODE REVIEW

File: src/main/java/se/sundsvall/disturbance/integration/db/SubscriptionRepository.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package se.sundsvall.disturbance.integration.db;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import se.sundsvall.disturbance.integration.db.model.SubscriptionEntity;
+
+public interface SubscriptionRepository extends JpaRepository<SubscriptionEntity, Long> {
+}
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/disturbance/integration/db/SubscriptionRepository.java:8]: `findByPartyId(String partyId)` ?

File: src/main/java/se/sundsvall/disturbance/integration/db/OptOutSettingsRepository.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package se.sundsvall.disturbance.integration.db;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import se.sundsvall.disturbance.integration.db.model.OptOutSettingsEntity;
+
+public interface OptOutSettingsRepository extends JpaRepository<OptOutSettingsEntity, Long> {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/disturbance/integration/db/OptOutSettingsRepository.java:7]: Do you need this interface?

File: src/main/java/se/sundsvall/disturbance/integration/db/SubscriptionRepository.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package se.sundsvall.disturbance.integration.db;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import se.sundsvall.disturbance.integration.db.model.SubscriptionEntity;
+
+public interface SubscriptionRepository extends JpaRepository<SubscriptionEntity, Long> {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/disturbance/integration/db/SubscriptionRepository.java:7]: No test class for this?

File: src/test/resources/db/scripts/schema.sql [hunk 4]
```diff
@@ -1,74 +1,108 @@
...
+                          created datetime(6),
+                          id bigint not null auto_increment,
+                          party_id varchar(255) not null,
+                          primary key (id)
+) engine=InnoDB;
+
+create table opt_out_settings (
+                                  id bigint not null auto_increment,
+                                  subscription_id bigint,
+                                  category enum ('COMMUNICATION','DISTRICT_COOLING','DISTRICT_HEATING','ELECTRICITY','ELECTRICITY_TRADE','WASTE_MANAGEMENT','WATER') not null,
```
Comment (foobar-1337) [inline src/test/resources/db/scripts/schema.sql:56]: Is it possible to make this `category varchar(255) not null` ?

File: src/main/java/se/sundsvall/disturbance/integration/db/OptOutSettingsRepository.java (hunk 2 above)
Comment (mha109) [reply src/main/java/se/sundsvall/disturbance/integration/db/OptOutSettingsRepository.java:7]: No I don't :)

File: src/test/resources/db/scripts/schema.sql (hunk 4 above)
Comment (mha109) [reply src/test/resources/db/scripts/schema.sql:56]: Yes!

**Review by foobar-1337 — DISMISSED** [review]

**Comment by foobar-1337** [comment]
Fix codesmells?

File: src/main/java/se/sundsvall/disturbance/service/SubscriptionService.java [hunk 5]
```diff
@@ -23,7 +39,20 @@ public Subscription create(final SubscriptionCreateRequest request) {
 		// throw Problem.valueOf(CONFLICT, format(ERROR_SUBSCRIPTION_ALREADY_EXISTS, request.getPartyId()));
 		// }
 
-		// TODO: map and save
+		//Temp-implementation
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/disturbance/service/SubscriptionService.java:42]: Remove?

File: src/test/java/se/sundsvall/disturbance/integration/db/SubscriptionRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	@Autowired
+	private SubscriptionRepository subscriptionRepository;
+
+	private static final String PARTY_ID = "partyId";
+
+	@Test
+	void testPersistAndfindByPartyId() {
+		//Setup
+		var subscriptionEntity = new SubscriptionEntity()
+				.withPartyId(PARTY_ID)
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/disturbance/integration/db/SubscriptionRepositoryTest.java:39]: Minor: 

1 tab for indentation (instead of 2)?

**Pull request merged**